### PR TITLE
[ws-man-bridge] Publish ws-instance-update - WEB-592

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Publishes the instance ID and workspace ID to redis. Currently there are no consumers, but we can observe metrics to verify the publishing works.

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-ws-man-d3e5aaf871</li>
	<li><b>🔗 URL</b> - <a href="https://mp-ws-man-d3e5aaf871.preview.gitpod-dev.com/workspaces" target="_blank">mp-ws-man-d3e5aaf871.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - mp-ws-man-redis-publish-instance-update-gha.12681</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/ws-manager-bridge/src/redis/publisher.ts [hunk 1]
```diff
@@ -7,19 +7,33 @@
...
 
     async publishPrebuildUpdate(): Promise<void> {
         log.debug("[redis] Publish prebuild udpate invoked.");
         this.metrics.reportUpdatePublished("prebuild");
     }
 
-    async publishInstanceUpdate(): Promise<void> {
-        log.debug("[redis] Publish instance udpate invoked.");
-        this.metrics.reportUpdatePublished("wsinstance");
+    async publishInstanceUpdate(update: WorkspaceInstanceUpdate): Promise<void> {
```
Comment (easyCZ) [inline components/ws-manager-bridge/src/redis/publisher.ts:25]: Will also add a test for this in this PR

File: components/ws-manager-bridge/src/redis/publisher.ts [hunk 2]
```diff
@@ -7,19 +7,33 @@
 import { inject, injectable } from "inversify";
 import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
 import { Metrics } from "../metrics";
+import { RedisClient } from "./client";
+import { WorkspaceInstanceUpdate, WorkspaceInstanceUpdatesChannel } from "@gitpod/gitpod-protocol";
```
Comment (geropl) [inline components/ws-manager-bridge/src/redis/publisher.ts:11]: I was somehow confused by the name `WorkspaceInstanceUpdate` and somehow expecing the full shape (with a similar name?) from ws-manager.
Can we somehow highlight (without making it to bulky) that this is for redis, or just the ids, or similar? :shrug:

Comment (easyCZ) [reply components/ws-manager-bridge/src/redis/publisher.ts:11]: I could prefix it with Redis: `RedisWorkspaceInstanceUpdate`

File: components/ws-manager-bridge/src/redis/publisher.ts (hunk 1 above)
Comment (easyCZ) [reply components/ws-manager-bridge/src/redis/publisher.ts]: added

File: components/ws-manager-bridge/src/redis/publisher.ts (hunk 2 above)
Comment (easyCZ) [reply components/ws-manager-bridge/src/redis/publisher.ts]: renamed

**Review by geropl — APPROVED** [review]
:heavy_check_mark:

**Pull request merged**